assessments/projects/openkruise/self-assessment.md

```diff
@@ -0,0 +1,563 @@
+are compiled into the same binary running on the API server. It consists of a
+`kruise-controller` and `kruise-webhook`.
+
+- `kruise-controller` - Responsible for checking the that the configuration is
+  in the desiredstate based on the user-configuration. It checks the resources
+  across all of the nodes and ensures they are up-to-date.
+
+- `kruise-webhook` - Used for admission control. It intercepts，validates and
+  potentially mutates requests coming from the `kruise-operator`. The
+  `kruise-webhook` is important since the `kube-apiserver` will fail the
+  request if the calling to the `kruise-webhook` fails.
+
+##### `kruise-daemon`
+
+This runs on every node and manages things like image pre-download and
+container restarting. It interacts with the `kruise-manager` indirectly by
+calling the API Server which calls through.
+
+##### `kube-apiserver`
+
+The Kubernetes API server validates and configures data for the api objects.
+This include pods, services, replicationcontrollers, and other things. It
+provides the frontend to the clusters shared state.
+
+##### `kubelet`
+
+The `kubelet` runs on each node. It's the primary "node-agent" that can register
+the node with the apiserver so that pods on that node can be managed.
+
+#### Actions
+
+These are the steps that a project performs in order to provide some service
+or functionality. These steps are performed by different actors in the system.
+Note, that an action need not be overly descriptive at the function call level.
+It is sufficient to focus on the security checks performed, use of sensitive
+data, and interactions between actors to perform an action.
+
+For example, the access server receives the client request, checks the format,
+validates that the request corresponds to a file the client is authorized to
+access, and then returns a token to the client. The client then transmits that
+token to the file server, which, after confirming its validity, returns the file.
+
+##### Operator based action
+
+One of the possible ways of interacting with the system are if a
+`kruise-operator` takes a manual action to administer some part of the system
+by using the CLI or a client SDK.
+
+###### Actors
+
+- `kruise-operator` - The person administrating or operating the system
+- `kubectl-kruise` - Allows user to interact with OpenKruise resources
+- `kruise-manager` - Processes user requests and oversees operations
+- `kruise-daemon` - Performs tasks at node level and monitors status of nodes
+
+###### Workflow
+
+1.  `kruise-operator` executes `kubctl-kruise` to initiate manual action
+    1.  Examples of manual action are scale and rollout
+2.  `kubctl-kruise` calls the `kruise-manager` which is part of the Control Plane
+3.  `kruise-daemon` calls the `kruise-manager` through the `kube-apiserver`
+4.  `kruise-daemon` executes an operation on the node
+    1.  Example include pre-image download and container restarting
```

Comment:
This is the action. The previous two sections (Operator based action & workflow) are not serving to increase clarity of the explanation, and can be omitted. Giving the action a title is fine, but the extra text only adds additional reading.

Also consider using plain english, to reduce the time it takes for readers to parse the bullets here.

Example (possibly inaccurate):

> When a user executes a `kubectl-kruise` command, such as `scale` or `rollout`, the CLI will call on _kruise-manager_, which is part of the control plane. Simultaneously, _kruise-daemon_ will call the _kruise manager_ through the _kube-apiserver_. Finally, _kruise-daemon_ will execute an operation on the node, such as to download an image or restart a container.

Reply:
The same suggestion also applies to the content below. Remove the fluff text and consider simplifying the bullets into clear action statements.

Reply:
I've gone ahead and only kept the two main actions which were described properly. They have been kept as "Manual command" and "InPlace Update". I've also removed the additional sections you mentioned which didn't help to increase the clarity of this section. This included the removal of the small section restating some of the actors already mentioned in the previous section.

Actions have also been rewritten from a bulleted-list to use more plain English with clear action statements. The example you provided above was indeed mostly accurate so I've used it with just a few minor changes when reworking the first action.